sdk/core/azure-core/azure/core/polling/_poller.py

```diff
@@ -266,7 +281,7 @@ def add_done_callback(self, func: Callable) -> None:
          argument, a completed LongRunningOperation.
         """
         # Still use "_done" and not "done", since CBs are executed inside the thread.
-        if self._done is None or self._done.is_set():
+        if not self._is_alive() or self._done.is_set():
             func(self._polling_method)
```

Comment:
It seems to me that this implementation (both original and my updates) has the potential to call a callback twice, both from this function in the main thread, and in the polling thread.
`self._done` is set before the `self._callbacks` are iterated, and if those callbacks take a while, then it's possible the supplied `func` could be called both here, and once we've added it callbacks, by the background thread.
Errors from the callbacks in the background thread are not propagated, so could be hard to debug.

Reply:
Seems that we could solve this by either:
- Remove the `self._done.is_set()` check - this would mean that new callbacks could be added to the background thread while existing callbacks are already running. This has the risk that we add a new callback just as the thread is finishing, and so it is never run.
- Only append `func` to `self._callbacks` in the scenario that we didn't run it here (i.e. add an `else` statement). This seems to be the most solid/reliable implementation, but removes the ability to access that callback after the fact (which I don't think we've had a need for yet anyway).

Reply:
I think the 2nd option makes more sense in the context of `remove_done_callback` - if we did append a new callback to `self._callbacks` after `self._done` is set, it cannot be removed as we raise a `ValueError`.
So I will probably add that fix unless anyone has concerns :)